### PR TITLE
ANE llama runner fixes for iOS26

### PR DESCRIPTION
**Pull request by metascroy**
This PR:

* Resolves https://github.com/pytorch/executorch/issues/15833.  The main issue is that iOS26 has a bug in its ANE implementation of SDPA, so we break SDPA into matmuls and softmax to avoid this.
* iOS26 also has large performance regressions over iOS18, so we tweak the model slightly (by exporting in parts) to achieve iOS18-level performance.
* We also add a C++ runner for the model split in parts.

With these changes, we estimate performance of Llama1B on iPhone 15 Pro / iOS 26 at:
* 30 tok/sec decode
* 1900 tok/sec prefill

Differential Revision: D88083155

### CODE REVIEW

File: examples/apple/coreml/llama/export.py [hunk 1]
```diff
@@ -102,67 +106,69 @@ def main() -> None:
...
-        replace_linear_with_split_linear(
-            model,
-            out_target_split_size=export_args.target_split_size,
-            out_max_splits=export_args.max_splits,
-            # I have not found splitting on in_features to be beneficial,
-            # and it often leads to OOM so I set in_max_splits to 1
-            in_target_split_size=1,
-            in_max_splits=1,
-        )
+    def maybe_split_model(model):
```
Comment (cccclai) [inline examples/apple/coreml/llama/export.py:121]: Is maybe_split_model mostly just split linear or something else

Comment (metascroy) [reply examples/apple/coreml/llama/export.py:121]: It splits the linear only.  But I have changes planned here.

**Comment by metascroy** [comment]
We are unifying iOS26 fixes and future development around the static model https://github.com/pytorch/executorch/pull/16184
